packages/hint-no-broken-links/tests/tests.ts

```diff
@@ -71,6 +75,20 @@ const tests: Array<HintTest> = [
         name: `This test should pass as it has an img with valid src value(absolute)`,
         serverConfig: generateHTMLPage('', bodyWithImageSource)
     },
+    {
+        name: `This test should pass as it has links with valid href values and a base tag which gets not used`,
+        serverConfig: {
+            '/': {content: generateHTMLPage('<base href="nested/">', bodyWithValidLinks)},
+            '/about': {content: 'My about page content'}
+        }
+    },
+    {
+        name: `This test should pass as it has a link a valid link (when resolved with the base tag)`,
+        serverConfig: {
+            '/': {content: generateHTMLPage('<base href="nested/">', bodyWithValidRelativeLink)},
+            '/nested/about': {content: 'My about page content'}
+        }
+    },
```

Comment:
Maybe add a test that fails? Something like:

```js
{
  name: `This test should fail because resolved URL with base doesn't exist `,
  reports: [{message: `Broken link found (404 response).` }],
  serverConfig: {
    '/': {content: generateHTMLPage('<base href="nested/">', bodyWithValidRelativeLink)},
    '/nested/about': {status: 404}
  }
}
```

I haven't tested the code above so there might be some modifications to do. There's more information in the [contributor guide](https://webhint.io/docs/contributor-guide/how-to/test-rules/). Also feel free to change the name of the test 😊